### ISSUE TITLE
Switching level causes an Atom crash

### ISSUE THREAD

**Issue by SelfishOlex** [o3de/o3de#8696]
**Describe the bug**
Calling LoadLevel the second time to a different level causes an Atom crash.

**Steps to reproduce**
1. create two levels, both can be empty levels
2. start a game launcher
3. loadlevel first level
4. loadlevel second level
5. See crash

**Expected behavior**
Second level loads to replace the first level

**Actual behavior**
Crashes
```
>	Atom_RPI.Private.dll!AZ::Job::Job(bool isAutoDelete, AZ::JobContext * context, bool isCompletion, char priority) Line 25	C++
 	Atom_RPI.Private.dll!AZ::JobCompletion::JobCompletion(AZ::JobContext * context) Line 28	C++
 	[Inline Frame] Atom_RPI.Private.dll!AZ::RPI::Scene::SimulateJobs() Line 433	C++
 	Atom_RPI.Private.dll!AZ::RPI::Scene::Simulate(AZ::RHI::JobPolicy jobPolicy, float simulationTime) Line 487	C++
 	Atom_RPI.Private.dll!AZ::RPI::RPISystem::SimulationTick() Line 279	C++
 	Atom_RPI.Private.dll!AZ::RPI::RPISystemComponent::OnSystemTick() Line 111	C++
 	[Inline Frame] MyProject.GameLauncher.exe!AZStd::Internal::INVOKE(void(AZ::SystemTickEvents::*)() &) Line 177	C++
 	[Inline Frame] MyProject.GameLauncher.exe!AZStd::invoke(void(AZ::SystemTickEvents::*)() &) Line 55	C++
 	[Inline Frame] MyProject.GameLauncher.exe!AZ::EBusEventProcessingPolicy::Call(void(AZ::SystemTickEvents::*)() &) Line 437	C++
 	MyProject.GameLauncher.exe!AZ::Internal::EBusContainer<AZ::SystemTickEvents,AZ::SystemTickEvents,0,1>::Dispatcher<AZ::EBus<AZ::SystemTickEvents,AZ::SystemTickEvents>>::Broadcast<void (__cdecl AZ::SystemTickEvents::*)(void)>(void(AZ::SystemTickEvents::*)() && func) Line 1360	C++
 	MyProject.GameLauncher.exe!AZ::ComponentApplication::TickSystem() Line 1357	C++

```
[crashlog_switching_levels.txt](https://github.com/o3de/o3de/files/8412311/crashlog_switching_levels.txt)

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU Ryzen 5900x
 - GPU NVidia RTX 3090
 - Memory 64GB

**Comment by santorac** [comment]
I was not able to repro this issue. (commit b38ca92ab44ec41b8f9e41fc21ec3932e84747b9). I used the AutomatedTesting project and switched back and forth between the Sponza and ShadowTest levels.

**Comment by superkitcath** [comment]
Hi @AMZN-Olex ! Can you confirm that this is still happening and what Game Project you are using? Thanks!

**Comment by LB-ArturZieba** [comment]
We have encountered a similar issue on the Stabilization_2205 (95dbd13) branch in a specific setup, on a specific project. Our issue presented when the Execute Console Command Script Canvas node was executed using the On Button Click Script Canvas node. In short when attempting to use the "loadlevel" command in that setup with a certain level (L1), the GameLauncher crashed on pressing the UI button set up with the described Script Canvas setup. It should be noted that executing the same setup but using a trigger for example, or attempting to load a different level would not cause this issue.

We did not find the root cause of the issue, but the described reproduction has a 100% reproduction rate. For more information refer to the attached error.zip (error.dmp), error.log, and Game.log from a sample crash. Additionally the project with the level (L1.prefab) that presents the issue is attached (GJ413.zip - has to be rebuilt in the Project Manager, and was originally made on the Source engine configuration).

If this should be logged as a separate issue, please let us know.

**Attached files:**
[error.zip](https://github.com/o3de/o3de/files/8521688/error.zip)
[Game.log](https://github.com/o3de/o3de/files/8521690/Game.log)
[error.log](https://github.com/o3de/o3de/files/8521691/error.log)
[GJ413.zip](https://github.com/o3de/o3de/files/8521694/GJ413.zip)

**Comment by VickyAtAZ** [comment]
@LB-ArturZieba I have tried with the provided 'GJ413' project and I got different behavior than a crash. 

I start the game launcher then use the lyshine UI menu to select the L1 level and it was loaded successfully. 
Then I pressed Esc key to show the UI menu then chose return to main menu. 
After that I got some asserts and the UI menu won't show anymore. 

@michabr The asserts I got were from the SpawnableEntitiesManager were trying to delete ui entities from game entity context.
